src/averageAges.js

```diff
@@ -20,6 +20,28 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  let result = 0;
+  const avAge = 0;
+  const man = people.filter(el => (el.sex === 'm'));
+
+  let menAvAge = man.reduce(
+    (accum, curr) => accum + (curr.died - curr.born),
+    avAge);
+
+  menAvAge /= man.length;
+
+  const centuryMan = man.filter(
+    el => (Math.ceil(el.died / 100) === century)
+  );
+  let centuryManAverage = centuryMan.reduce(
+    (accum, curr) => accum + (curr.died - curr.born),
+    avAge);
+
+  centuryManAverage /= centuryMan.length;
+
+  (century === undefined) ? result = menAvAge : result = centuryManAverage;
```

Comment:
Don't repeat yourself: method reduce and calculating average age is the same for both of variants: will be better to check at the start of the function, if century is undefined - you should work with array of all men
if century is not undefined - you should work with array of men with correct century